transformer_engine/pytorch/utils.py

```diff
@@ -33,7 +31,7 @@ def clear_tensor_data(*tensors: Tuple[Optional[torch.Tensor], ...]) -> None:
     """
     for t in tensors:
         if t is not None:
-            if isinstance(t, QuantizedTensor):
+            if hasattr(t, "clear"):
```

Comment:
Nit: I understand why this change was made but it would be robust would be to check for each base type separately and not for this attribute which PyTorch could add in the future to their tensors. Maybe we can come up with some common dummy base class for all the _internal BaseTensors for these types of situations? @ptrendx